src/Data/API/Tools/QuickCheck.hs

```
@@ -19,7 +19,9 @@ import           Test.QuickCheck                as QC
 import           Prelude
 
 
--- | Tool to generate 'Arbitrary' instances for generated types.
+-- | Tool to generate 'Arbitrary' instances for generated types. This tool generates
+-- also a stock shrinker via the 'generic-arbitrary' package, which means we require
+-- the wrapped type to be an instance of 'Generic'.
```

Now that I think about it, maybe we need to extend this to check if we have a `Generic` instance in scope and, if we have then generate the default implementation, to not force `Generic` over the client?

Alternatively, as the former plan can silently discard instances, we could even create a separate tool, albeit that doesn't seem clean?